torch/distributed/distributed_c10d.py

```diff
@@ -1446,6 +1457,8 @@ def _new_process_group_helper(
             _, prefix_store = _world.pg_map[existing_group]
             return existing_group, prefix_store
 
+    group_desc = "unknown" if group_desc is None else group_desc
```

Comment:
i think "unknown" is the best we can do, even though i don't love it.

at least, when we create groups as part of an N-D device-mesh, we should make sure they get named according to the device-mesh dims automatically.  cc @wz337

Reply:
this reminds me that we should also add support for new_subgroups and similar funcs that the subgroup should inherit a group_desc.

Reply:
Is there a preference of using "unknown" over ""?
If pytorch internal coding (or even external usage bc I see the PR added a property API),
`if group_desc` seems easier than `if group_desc == "unknown"`.